Code/Framework/AzCore/AzCore/IO/Streamer/DedicatedCache.h

```diff
@@ -11,84 +11,88 @@
-
-            ~DedicatedCacheConfig() override = default;
-            AZStd::shared_ptr<StreamStackEntry> AddStreamStackEntry(
-                const HardwareInformation& hardware, AZStd::shared_ptr<StreamStackEntry> parent) override;
-            static void Reflect(AZ::ReflectContext* context);
-
-            //! The size of the individual blocks inside the cache.
-            BlockCacheConfig::BlockSize m_blockSize{ BlockCacheConfig::BlockSize::MemoryAlignment };
-            //! The overall size of the cache in megabytes.
-            u32 m_cacheSizeMib{ 8 };
-            //! If true, only the epilog is written otherwise the prolog and epilog are written. In either case both prolog and epilog are read.
-            //! For uses of the cache that read mostly sequentially this flag should be set to true. If reads are more random than it's better
-            //! to set this flag to false.
-            bool m_writeOnlyEpilog{ true };
-        };
-
-        class DedicatedCache
-            : public StreamStackEntry
-        {
-        public:
-            DedicatedCache(u64 cacheSize, u32 blockSize, u32 alignment, bool onlyEpilogWrites);
-            
-            void SetNext(AZStd::shared_ptr<StreamStackEntry> next) override;
-            void SetContext(StreamerContext& context) override;
-
-            void PrepareRequest(FileRequest* request) override;
-            void QueueRequest(FileRequest* request) override;
-            bool ExecuteRequests() override;
-
-            void UpdateStatus(Status& status) const override;
-
-            void UpdateCompletionEstimates(AZStd::chrono::system_clock::time_point now, AZStd::vector<FileRequest*>& internalPending,
-                StreamerContext::PreparedQueue::iterator pendingBegin, StreamerContext::PreparedQueue::iterator pendingEnd) override;
-
-            void CollectStatistics(AZStd::vector<Statistic>& statistics) const override;
-
-        private:
-            void CreateDedicatedCache(FileRequest* request, FileRequest::CreateDedicatedCacheData& data);
-            void DestroyDedicatedCache(FileRequest* request, FileRequest::DestroyDedicatedCacheData& data);
-
-            void ReadFile(FileRequest* request, FileRequest::ReadData& data);
-            size_t FindCache(const RequestPath& filename, FileRange range);
-            size_t FindCache(const RequestPath& filename, u64 offset);
-
-            void FlushCache(const RequestPath& filePath);
-            void FlushEntireCache();
-
-            AZStd::vector<RequestPath> m_cachedFileNames;
-            AZStd::vector<FileRange> m_cachedFileRanges;
-            AZStd::vector<AZStd::unique_ptr<BlockCache>> m_cachedFileCaches;
-            AZStd::vector<size_t> m_cachedFileRefCounts;
-
-            AZ::Statistics::RunningStatistic m_usagePercentageStat;
+        struct CreateDedicatedCacheData;
+        struct DestroyDedicatedCacheData;
+    } // namespace Requests
+
+    struct DedicatedCacheConfig final : public IStreamerStackConfig
```

Comment:
It is preferred to follow the "chameleon rule" when making changes regard the style of a file to avoid making unnecessary changes.
So the base class moving from a separate line to a line on it's own seems kind of strange.

Reply:
I understand the rule, the clang-format is fighting me though :smile: 

Maybe we should consider running clang-format on all the source files to prevent such things?

Reply:
Yeah we totally should, some files have it and some do not, for those that do not the best way is to just select the updated lines and use `Ctrl-K, F`. It might be worth creating a format-only PR (for a small number of files at a time) to update things to stop functional and form changes getting mixed in together